snuba/datasets/sessions.py

```diff
@@ -0,0 +1,270 @@
+            ]
+        )
+        read_schema = AggregatingMergeTreeSchema(
+            columns=read_columns,
+            local_table_name=READ_LOCAL_TABLE_NAME,
+            dist_table_name=READ_DIST_TABLE_NAME,
+            prewhere_candidates=["project_id", "org_id"],
+            order_by="(org_id, project_id, started, release)",
+            partition_by="(toMonday(started))",
+            settings={"index_granularity": 16384},
+        )
+        materialized_view = MaterializedViewSchema(
+            local_materialized_view_name=READ_LOCAL_MV_NAME,
+            dist_materialized_view_name=READ_DIST_MV_NAME,
+            prewhere_candidates=["project_id", "org_id"],
+            columns=read_columns,
+            query="""
+                SELECT
+                    org_id,
+                    project_id,
+                    toStartOfHour(started) as started,
+                    release,
+                    environment,
+                    quantilesIfState(0.5, 0.9)(
+                        duration,
+                        duration <> 4294967295 AND status == 1
+                    ) as duration,
+                    countIfState(session_id, seq == 0) as uniq_sessions,
+                    uniqIfState(distinct_id, distinct_id != '00000000-0000-0000-0000-000000000000') as uniq_users,
+                    uniqIfState(session_id, status == 2) as uniq_sessions_crashed,
+                    uniqIfState(session_id, status == 3) as uniq_sessions_abnormal,
+                    uniqIfState(session_id, errors > 0) as uniq_sessions_errored,
+                    uniqIfState(distinct_id, status == 2) as uniq_users_crashed,
+                    uniqIfState(distinct_id, status == 3) as uniq_users_abnormal,
+                    uniqIfState(distinct_id, errors > 0) as uniq_users_errored
+                FROM
+                    %(source_table_name)s
+                WHERE
+                    deleted == 0
+                GROUP BY
+                    org_id, project_id, started, release, environment
+            """,
+            local_source_table_name=WRITE_LOCAL_TABLE_NAME,
+            local_destination_table_name=READ_LOCAL_TABLE_NAME,
+            dist_source_table_name=WRITE_DIST_TABLE_NAME,
+            dist_destination_table_name=READ_DIST_TABLE_NAME,
+        )
+
+        dataset_schemas = DatasetSchemas(
+            read_schema=read_schema,
+            write_schema=write_schema,
+            intermediary_schemas=[materialized_view],
+        )
+
+        table_writer = TableWriter(
+            write_schema=write_schema,
+            stream_loader=KafkaStreamLoader(
+                processor=SessionsProcessor(), default_topic="ingest-sessions",
+            ),
+        )
+
+        super().__init__(
+            dataset_schemas=dataset_schemas,
+            table_writer=table_writer,
+            time_group_columns={"bucketed_started": "started"},
```

Comment:
Thinking over this, I don't think you need the TimeSeriesDataset with a bucketed column if you GROUP BY toStartOfHour(timestamp) when generating the concrete table. Which you should probably do otherwise we are not aggregating by hour but by second.

Reply:
Going to revisit this. I believe it makes queries easier still.